lib/galaxy/authnz/custos_authnz.py

```diff
@@ -0,0 +1,212 @@
+        email = userinfo['email']
+        user_id = userinfo['sub']
+
+        # Create or update custos_authnz_token record
+        custos_authnz_token = self._get_custos_authnz_token(trans.sa_session, user_id, self.config['provider'])
+        if custos_authnz_token is None:
+            user = self._get_current_user(trans)
+            if not user:
+                user = self._create_user(trans.sa_session, username, email)
+            custos_authnz_token = CustosAuthnzToken(user=user,
+                                   external_user_id=user_id,
+                                   provider=self.config['provider'],
+                                   access_token=access_token,
+                                   id_token=id_token,
+                                   refresh_token=refresh_token,
+                                   expiration_time=expiration_time,
+                                   refresh_expiration_time=refresh_expiration_time,
+                                   raw_token=token)
+        else:
+            custos_authnz_token.access_token = access_token
+            custos_authnz_token.id_token = id_token
+            custos_authnz_token.refresh_token = refresh_token
+            custos_authnz_token.expiration_time = expiration_time
+            custos_authnz_token.refresh_expiration_time = refresh_expiration_time
+            custos_authnz_token.raw_token = token
+        trans.sa_session.add(custos_authnz_token)
+        trans.sa_session.flush()
+        return login_redirect_url, custos_authnz_token.user
+
+    def disconnect(self, provider, trans, disconnect_redirect_url=None):
+        try:
+            user = trans.user
+            # Find CustosAuthnzToken record for this provider (should only be one)
+            provider_tokens = [token for token in user.custos_auth if token.provider == self.config["provider"]]
+            if len(provider_tokens) == 0:
+                raise Exception("User is not associated with provider {}".format(self.config["provider"]))
+            if len(provider_tokens) > 1:
+                raise Exception("User is associated more than once with provider {}".format(self.config["provider"]))
+            trans.sa_session.delete(provider_tokens[0])
+            trans.sa_session.flush()
+            return True, "", disconnect_redirect_url
+        except Exception as e:
+            return False, "Failed to disconnect provider {}: {}".format(provider, str(e)), None
+
+    def _create_oauth2_session(self, state=None, scope=None):
+        client_id = self.config['client_id']
+        redirect_uri = self.config['redirect_uri']
+        if (redirect_uri.startswith('http://localhost')
+                and os.environ.get("OAUTHLIB_INSECURE_TRANSPORT", None) != "1"):
+            log.warn("Setting OAUTHLIB_INSECURE_TRANSPORT to '1' to "
+                     "allow plain HTTP (non-SSL) callback")
+            os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = "1"
```

Comment:
No because VERIFY_SSL is used to control whether to verify SSL of the authorization server whereas OAUTHLIB_INSECURE_TRANSPORT is used to enable insecure callback on the client server.